onnxruntime/core/optimizer/graph_optimizer.cc

```diff
@@ -0,0 +1,181 @@
+
+#include "core/common/common.h"
+#include "core/common/status.h"
+#include "core/common/logging/logging.h"
+#include "core/common/logging/macros.h"
+#include "core/framework/tensorprotoutils.h"
+#include "core/framework/kernel_registry.h"
+#include "core/framework/data_types.h"
+#include "core/framework/mldata_type_utils.h"
+#include "core/graph/graph_viewer.h"
+#include "core/optimizer/graph_optimizer.h"
+
+namespace onnxruntime {
+
+// Build the MLValue name->idx mapping
+common::Status GraphOptimizer::InitMLValueNameIndexMapping() {
```

Comment:
>InitMLValueNameIndexMapping [](start = 31, length = 27)

This should be in common code (e.g. helper method in framework utils.h/.cc) and used by both here and SessionStateInitializer::SaveMLValueNameIndexMapping instead of cut-and-pasted.